### ISSUE TITLE
No ports exposed in image?

### ISSUE THREAD

**Issue by mxchinegod** [obsidiandynamics/kafdrop#460]

**Comment by brsolomon-deloitte** [comment]
not stale

**Comment by brsolomon-deloitte** [comment]
not stale

**Comment by davideicardi** [comment]
Can someone provide a better explanation of the problem and suggestions on how to fix it?

**Comment by brsolomon-deloitte** [comment]
The app's Dockerfile could use [`EXPOSE`](https://docs.docker.com/engine/reference/builder/#expose).

**Comment by Bert-R** [comment]
Suggestion: as Davide merely looks after Kafdrop for lack of maintainers, help him by:
* Extending the ticket description with the benefits of adding ``EXPOSE`` to the ``Dockerfile``
* Creating a PR that adds the ``EXPOSE`` instruction

**Comment by mehdihasan** [comment]
According to the [Docker documentation](https://docs.docker.com/engine/reference/builder/#expose), 

> the `EXPOSE` instruction informs Docker that the container listens on the specified network ports at runtime. 

And then,

> The EXPOSE instruction does not actually publish the port. It functions as a type of documentation between the person who builds the image and the person who runs the container, about which ports are intended to be published. To actually publish the port when running the container, use the -p flag on docker run to publish and map one or more ports, or the -P flag to publish all exposed ports and map them to high-order ports.

Some say `EXPOSE` is used to use for inter container communication - in cases the application does not need to exposed to the host machine, i.e. a communication between an application container and a Redis container running in the same host. However, actually one does not need to use the `EXPOSE` keyword for inter container networking.

So, basically `EXPOSE` is nothing but a contract keyword between the dev and the ops, i.e. dev is telling the ops that this is the port I want to EXPOSE my app.

In case of Kafdrop, it can be added in the [`Dockerfile`](https://github.com/obsidiandynamics/kafdrop/blob/master/src/main/docker/Dockerfile)  as following:

```
FROM eclipse-temurin:11.0.18_10-jdk

ADD kafdrop.sh /
ADD kafdrop*tar.gz /

RUN chmod +x /kafdrop.sh

EXPOSE 9000

ENTRYPOINT ["/kafdrop.sh"]
```

Please feel free to correct me if I am wrong.
